cmd/client/client.go

```diff
@@ -0,0 +1,94 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"os"
+)
+
+// RoomID represents the room ID
+type RoomID string
+
+type chatClient interface {
+	Run()
```

Comment:
Returning error is better.
What if error is raised when the run is starting (and not when chatting), in such cases we would like to let the client know there is a problem(main.go should know how to handle such cases).
For me, functions not returning any value are bad habit unless output is written to a channel or something like that :)